presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -130,9 +131,9 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
-    private Estimate calculateNullsCount(Map<String, PartitionStatistics> statisticsByPartitionName, String column)
+    private Estimate calculateNullsFraction(Map<String, PartitionStatistics> statisticsByPartitionName, String column, Estimate totalRowsCount)
     {
-        return summarizePartitionStatistics(
+        Estimate totalNullsCount = summarizePartitionStatistics(
```

Comment:
This function call is overly complicated to me, I would refactor that. Of course as a follow up.